action.yaml

```diff
@@ -176,27 +192,37 @@ runs:
       shell: bash
       run: |
         n=0
-        until [ "$n" -ge 30 ]; do
-          started=1
+        started=1
+        until [ "$n" -ge ${{ inputs.ssh-startup-wait-retries }} ]; do
           if grep -E ".*OK.*Started.*ssh.*" /tmp/console.log; then
-            break || started=0
+            started=0
+            break
           elif grep -E ".*FAILED.*Failed.*to.*start.*ssh*" /tmp/console.log; then
-            cat /tmp/console.log && exit 40
+            cat /tmp/console.log
+            exit 40
           fi
           n=$((n+1))
-          sleep 1
+          sleep ${{ inputs.ssh-startup-wait-timeout }}
         done
-        [ $started -eq 1 ] || (cat /tmp/console.log && exit 41)
-
+        if [ $started -eq 1 ]; then
+          cat /tmp/console.log
+          exit 41
+        fi
+        
         n=0
-        until [ "$n" -ge 5 ]; do
-          success=1
-          ssh -p ${{ inputs.ssh-port }} -o "StrictHostKeyChecking=no" root@localhost exit && \
-          break || success=0
+        success=1
+        until [ "$n" -ge ${{ inputs.ssh-connect-wait-retries }} ]; do
+          if ssh -p ${{ inputs.ssh-port }} -o "StrictHostKeyChecking=no" root@localhost exit; then
+            success=0
```

Comment:
nit: `success=0` means success and `succuess=1` means failure? This is somewhat counterintuitive.